lib/std/fs/test.zig

```diff
@@ -907,6 +907,16 @@ test "makePath in a directory that no longer exists" {
     try testing.expectError(error.FileNotFound, tmp.dir.makePath("sub-path"));
 }
 
+test "makePath but sub_path contains pre-existing file" {
```

Comment:
If you rebase on master, the WASI failure should be fixed (https://github.com/ziglang/zig/pull/17029), but it'll start hitting https://github.com/ziglang/zig/issues/17034 so you'll need to add this conditional to the test case:

```suggestion
test "makePath but sub_path contains pre-existing file" {
    // TODO: Re-enable once https://github.com/ziglang/zig/issues/17034 is solved
    if (builtin.os.tag == .linux and builtin.link_libc and builtin.abi == .gnu) return error.SkipZigTest;

```